dev/com.ibm.ws.jca_fat_regr/fat/src/com/ibm/ws/jca/fat/regr/util/JCAFATTest.java

```diff
@@ -0,0 +1,235 @@
+package com.ibm.ws.jca.fat.regr.util;
+
+/*******************************************************************************
+ * Copyright (c) 2012, 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+import static org.junit.Assert.fail;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.junit.Rule;
+import org.junit.rules.TestName;
+import org.junit.runner.Description;
+
+import com.ibm.websphere.simplicity.RemoteFile;
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.topology.impl.LibertyServer;
+
+/**
+ * Common infrastructure code for JCA REGR FAT tests
+ */
+public abstract class JCAFATTest {
```

Comment:
Can this class extend `FATServletClient`? 
This will reduce some repeated code that could be stripped out.

Reply:
made this change